src/main/java/net/glowstone/net/handler/play/inv/WindowClickHandler.java

```diff
@@ -42,7 +42,7 @@ private boolean process(final GlowPlayer player, final WindowClickMessage messag
         final InventoryView view = player.getOpenInventory();
         final GlowInventory top = (GlowInventory) view.getTopInventory();
         final GlowInventory bottom = (GlowInventory) view.getBottomInventory();
-        final ItemStack slotItem = view.getItem(viewSlot);
+        final ItemStack slotItem = message.getItem();
```

Comment:
Yeah, and the underlying issue was fixed in https://github.com/GlowstonePlusPlus/GlowstonePlusPlus/commit/0be6bc0a5ec62dbf8aa967d831c7a1c47e6536fc

I had followed a plugin stacktrace incorrectly, as I was trying to fix a NullPointerException error with many plugins that deal with items. I discovered it was because these plugins were expecting air, not null, probably because of a Bukkit rule. So I "fixed" getItem, instead of fixing the true culprit, getItemInHand, which was fixed in above mentioned commit. The previous "fix" is what fixed a mostly stable crafting.

And your contribution probably made it perfect.